Format and complete README (#99)
sensu-plugins/sensu-plugins-memory-checks: README.md

[![Gem Version](https://badge.fury.io/rb/sensu-plugins-memory-checks.svg)](http://badge.fury.io/rb/sensu-plugins-memory-checks)
[![Code Climate](https://codeclimate.com/github/sensu-plugins/sensu-plugins-memory-checks/badges/gpa.svg)](https://codeclimate.com/github/sensu-plugins/sensu-plugins-memory-checks)
[![Test Coverage](https://codeclimate.com/github/sensu-plugins/sensu-plugins-memory-checks/badges/coverage.svg)](https://codeclimate.com/github/sensu-plugins/sensu-plugins-memory-checks)
[![Dependency Status](https://gemnasium.com/sensu-plugins/sensu-plugins-memory-checks.svg)](https://gemnasium.com/sensu-plugins/sensu-plugins-memory-checks)

## Sensu Memory Check Plugin

- [Overview](#overview)
- [Files](#files)
- [Usage examples](#usage-examples)
- [Configuration](#configuration)
  - [Sensu Go](#sensu-go)
    - [Asset registration](#asset-registration)
    - [Asset definition](#asset-definition)
    - [Check definition](#check-definition)
  - [Sensu Core](#sensu-core)
    - [Check definition](#check-definition)
- [Installation from source](#installation-from-source)
- [Additional notes](#additional-notes)
- [Contributing](#contributing)

### Overview

> This plugin provides native disk instrumentation for monitoring and metrics collection, including: health, usage, and various metrics.

Reviewer: ```suggestion
```

This seems like an accidental addition intended for the disk-checks collection?

```

#### Check definition

```yaml
---
type: CheckConfig
spec:
  command: "check-memory.rb"
  handlers: []
  high_flap_threshold: 0
  interval: 10
  low_flap_threshold: 0
  publish: true
  runtime_assets:
  - sensu-plugins/sensu-plugins-memory-checks
  - sensu/sensu-ruby-runtime
  subscriptions:
  - linux
```

### Sensu Core

#### Check definition
```json
{
  "checks": {
    "metrics-disk-usage": {
      "command": "metric-disk-usage.rb",
      "subscribers": ["linux"],
      "interval": 10,
      "refresh": 10,
      "handlers": ["influxdb"]
    }
  }
}
```

## Installation from source

### Sensu Go

See the instructions above for [asset registration](#asset-registration)

### Sensu Core

Install and setup plugins on [Sensu Core](https://docs.sensu.io/sensu-core/latest/installation/installing-plugins/)

## Additional notes

### `bin/check-ram.rb`

Ruby does not have a good native way to grab information on memory usage (without using C extensions or running shell commands and parsing out). For this reason, `check-ram.rb` uses a gem called `vmstat` that has a dependency on a GCC to compile the C extensions.

> To avoid errors which occur when trying to compile the C extensions without GCC present, we do not install `vmstat` gem by default.

Author: ```suggestion
To avoid errors that occur when trying to compile the C extensions without GCC present, we do not install the `vmstat` gem by default.
```

> **Users who need to use `check-ram.rb` must install build tools including GCC**. This usually requires two steps but can vary based on your distribution and Ruby setup:

Author: ```suggestion
**Users who need to use `check-ram.rb` must install build tools, including GCC**. This usually requires two steps but can vary based on your distribution and Ruby setup:
```

```

#### Check definition

```yaml
---
type: CheckConfig
spec:
  command: "check-memory.rb"
  handlers: []
  high_flap_threshold: 0
  interval: 10
  low_flap_threshold: 0
  publish: true
  runtime_assets:
  - sensu-plugins/sensu-plugins-memory-checks
  - sensu/sensu-ruby-runtime
  subscriptions:
  - linux
```

### Sensu Core

#### Check definition
```json
{
  "checks": {
    "metrics-disk-usage": {
      "command": "metric-disk-usage.rb",
      "subscribers": ["linux"],
      "interval": 10,
      "refresh": 10,
      "handlers": ["influxdb"]
    }
  }
}
```

## Installation from source

### Sensu Go

See the instructions above for [asset registration](#asset-registration)

### Sensu Core

Install and setup plugins on [Sensu Core](https://docs.sensu.io/sensu-core/latest/installation/installing-plugins/)

## Additional notes

### `bin/check-ram.rb`

Ruby does not have a good native way to grab information on memory usage (without using C extensions or running shell commands and parsing out). For this reason, `check-ram.rb` uses a gem called `vmstat` that has a dependency on a GCC to compile the C extensions.

> Some users may not need to use `check-ram.rb`. These users do not need gcc to use the rest of this plugin, so we do not install GCC by default.

Reviewer: ```suggestion
To avoid errors which occur when trying to compile the C extensions without GCC present, we do not install `vmstat` gem by default.
```

> **Users who need to use `check-ram.rb` must install GCC**. GCC installation usually requires two steps but can vary based on your distribution and Ruby setup:

Reviewer: ```suggestion
**Users who need to use `check-ram.rb` must install build tools including GCC**. This usually requires two steps but can vary based on your distribution and Ruby setup:
```

> 1. Install `gcc`. If you are on a Debian system, GCC is located in the `build-essential` package.

Reviewer: ```suggestion
1. Install `gcc`. If you are on a Debian system, GCC is located in the `build-essential` package. On RHEL systems, `yum groupinstall "Development tools" ` will provide the required packages.
```
